Return name of layout in the templates.list endpoint

WouterSioen opened the pull request:
This is necessary to visual display templates.

WouterSioen committed:
Return name of layout in the templates.list endpoint

JordyDeGroeve commented on src/01-general/document-templates.apib:
```diff
@@ -38,3 +38,4 @@ Get a list of all document templates.
                         + timetracking_report
                         + workorder
                 + is_default: true (boolean)
+                + name: `new logo` (string)
```
B: this should be a `nullable` string. We don't have a name for the default variations. We should return `null` in this case

WouterSioen committed:
Make clear that template name can be null

The default variation does not have a name. In the web UI, this falls
back to the department, but this is not the "real" name of that
variation, so let's return null. This way every integration can choose
how to display this without us forcing something upon them.

JordyDeGroeve commented:
Q: should we add a comment in which case that we return `null`? Or is this not needed?

WouterSioen commented:
I don't feel like that's necessary?
